policy.go

```diff
@@ -69,61 +67,131 @@ type policyOrDataTree []policyOrDataNode
+	dynamicPolicyRef       tpm2.Nonce
+}
+
+// staticPolicyDataRaw_v0 is version 0 of the on-disk format of staticPolicyData.
+type staticPolicyDataRaw_v0 struct {
 	AuthPublicKey        *tpm2.Public
 	PinIndexHandle       tpm2.Handle
 	PinIndexAuthPolicies tpm2.DigestList
 }
 
-// staticPolicyDataRaw_v0 is version 0 of the on-disk format of staticPolicyData. They are currently the same structures.
-type staticPolicyDataRaw_v0 staticPolicyData
-
 func (d *staticPolicyDataRaw_v0) data() *staticPolicyData {
-	return (*staticPolicyData)(d)
+	return &staticPolicyData{
+		authPublicKey:          d.AuthPublicKey,
+		policyCounterHandle:    d.PinIndexHandle,
+		v0PinIndexAuthPolicies: d.PinIndexAuthPolicies}
 }
 
-// makeStaticPolicyDataRaw_v0 converts staticPolicyData to version 0 of the on-disk format. They are currently the same structures
-// so this is just a cast, but this may not be the case if the metadata version changes in the future.
+// makeStaticPolicyDataRaw_v0 converts staticPolicyData to version 0 of the on-disk format.
 func makeStaticPolicyDataRaw_v0(data *staticPolicyData) *staticPolicyDataRaw_v0 {
-	return (*staticPolicyDataRaw_v0)(data)
+	return &staticPolicyDataRaw_v0{
+		AuthPublicKey:        data.authPublicKey,
+		PinIndexHandle:       data.policyCounterHandle,
+		PinIndexAuthPolicies: data.v0PinIndexAuthPolicies}
+}
+
+// staticPolicyDataRaw_v1 is version 1 of the on-disk format of staticPolicyData.
+type staticPolicyDataRaw_v1 struct {
+	AuthPublicKey       *tpm2.Public
+	PolicyCounterHandle tpm2.Handle
+	DynamicPolicyRef    tpm2.Nonce
+}
+
+func (d *staticPolicyDataRaw_v1) data() *staticPolicyData {
+	return &staticPolicyData{
+		authPublicKey:       d.AuthPublicKey,
+		policyCounterHandle: d.PolicyCounterHandle,
+		dynamicPolicyRef:    d.DynamicPolicyRef}
+}
+
+// makeStaticPolicyDataRaw_v1 converts staticPolicyData to version 1 of the on-disk format.
+func makeStaticPolicyDataRaw_v1(data *staticPolicyData) *staticPolicyDataRaw_v1 {
+	return &staticPolicyDataRaw_v1{
+		AuthPublicKey:       data.authPublicKey,
+		PolicyCounterHandle: data.policyCounterHandle,
+		DynamicPolicyRef:    data.dynamicPolicyRef}
+}
+
+// computeDynamicPolicyCounterAuthPolicies computes the authorization policy digests passed to TPM2_PolicyOR for a dynamic
+// authorization policy counter that can be updated with the key associated with updateKeyName.
+func computeDynamicPolicyCounterAuthPolicies(alg tpm2.HashAlgorithmId, updateKeyName tpm2.Name) (tpm2.DigestList, error) {
+	// The NV index requires 2 policies:
+	// - A policy to initialize the index with no authorization
+	// - A policy for updating the index to revoke old dynamic authorization policies, using a signed assertion.
+	var authPolicies tpm2.DigestList
+
+	trial, err := tpm2.ComputeAuthPolicy(alg)
+	if err != nil {
+		return nil, err
+	}
+	trial.PolicyNvWritten(false)
+	authPolicies = append(authPolicies, trial.GetDigest())
+
+	trial, _ = tpm2.ComputeAuthPolicy(alg)
```

Comment:
Why is the error return discarded in this case?

Reply:
The only way this can fail is if the algorithm isn't supported by the go runtime, but there's an earlier call to tpm2.ComputeAuthPolicy in this function with the same algorithm where the error is checked. If that succeeded then there isn't a way for this call to fail.